[Automation] native-image run fails for org.flywaydb:flyway-sqlserver:11.12.0

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=11.12.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.flywaydb:flyway-sqlserver:11.8.1`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27047956571
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
urces@file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/resources/test/META-INF/native-image/reachability-metadata.json+api=META-INF/native-image/reachability-metadata.json \
-H:FallbackThreshold@user+api=0 \
-H:Name@user+api=org.flywaydb.flyway-sqlserver_tests-tests \
-H:ConfigurationFileDirectories@user=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.awaitility/awaitility/4.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest/2.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.flywaydb/flyway-sqlserver/11.8.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.flywaydb/flyway-core/12.5.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-databind/2.15.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-annotations/2.3.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-core/2.6.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.microsoft.sqlserver/mssql-jdbc/12.2.0.jre11 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:+AddAllCharsets@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=78beca1b-411b-ca03-928a-b9267f008db0 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
========================================================================================================================
GraalVM Native Image: Generating 'org.flywaydb.flyway-sqlserver_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (6.3s @ 0.19GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [******]                                                                  (48.9s @ 1.13GB)
   12,524 types,  16,055 fields, and  58,742 methods found reachable
    4,127 types,     697 fields, and   4,434 methods registered for reflection
       63 types,      71 fields, and      55 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (5.6s @ 1.32GB)
[4/8] Parsing methods...      [**]                                                                       (4.8s @ 1.42GB)
[5/8] Inlining methods...     [***]                                                                      (2.7s @ 1.54GB)
[6/8] Compiling methods...    [*****]                                                                   (26.8s @ 1.06GB)
[7/8] Laying out methods...   [**]                                                                       (4.5s @ 1.42GB)
[8/8] Creating image...       [**]                                                                       (4.7s @ 1.60GB)
  18.36MB (36.42%) for code area:    37,662 compilation units
  26.21MB (52.00%) for image heap:  322,556 objects and 168 resources
   5.84MB (11.58%) for other data
  50.41MB in total image size, 47.12MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   8.24MB java.base                                            5.53MB byte[] for code metadata
   3.33MB svm.jar (Native Image)                               4.61MB char[]
   1.36MB mssql-jdbc-12.2.0.jre11.jar                          4.45MB byte[] for java.lang.String
 952.16kB jackson-databind-2.19.1.jar                          2.06MB java.lang.String
 596.22kB flyway-core-11.12.0.jar                              1.61MB java.lang.Class
 556.55kB assertj-core-3.22.0.jar                              1.30MB com.oracle.svm.core.hub.DynamicHubCompanion
 536.72kB java.xml                                             1.29MB byte[] for general heap data
 290.60kB jackson-core-2.19.1.jar                            476.42kB java.util.concurrent.ConcurrentHashMap$Node
 236.46kB java.rmi                                           430.48kB byte[] for reflection metadata
 198.63kB junit-jupiter-engine-5.9.2.jar                     425.86kB java.util.HashMap$Node
   1.47MB for 38 more packages                                 4.03MB for 3191 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 25 component(s) is embedded in binary (1.28kB). 2 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       8.7s (8.1% of total time) in 1162 GCs | Peak RSS: 2.43GB | CPU load: 3.64
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/org.flywaydb.flyway-sqlserver_tests-tests (executable)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'org.flywaydb.flyway-sqlserver_tests-tests' in 1m 46s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile

> Task :nativeTest
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------
Jun 06, 2026 1:41:35 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 918e2718-16e9-462d-bbe1-0cde78e984ce Prelogin error: host localhost port 1433 Unexpected end of prelogin response after 0 bytes read
Jun 06, 2026 1:41:35 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 0a3fc598-b4d2-4fba-bd13-627d4cf36f79 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:0a3fc598-b4d2-4fba-bd13-627d4cf36f79
Jun 06, 2026 1:41:35 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: ff331c76-1a9c-4345-9854-b7b4e956265e Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:ff331c76-1a9c-4345-9854-b7b4e956265e
Jun 06, 2026 1:41:36 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: bd7d4770-bec8-48d6-99df-ebd7f06f3731 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:bd7d4770-bec8-48d6-99df-ebd7f06f3731

Starting MSSQL ...
Jun 06, 2026 1:41:37 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: fe8bb297-beb6-45ba-b444-902cfe0ac38d Prelogin error: host localhost port 1433 Unexpected end of prelogin response after 0 bytes read
Jun 06, 2026 1:41:38 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 37a8e014-15ef-4c48-a84a-8d8a7c2f22e4 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:37a8e014-15ef-4c48-a84a-8d8a7c2f22e4
MSSQL started
Shutting down MSSQL
flyway.sqlserver.FlywaySqlServerTests > migrate() FAILED


Failures (1):
  JUnit Jupiter:FlywaySqlServerTests:migrate()
    MethodSource [className = 'flyway.sqlserver.FlywaySqlServerTests', methodName = 'migrate', methodParameterTypes = '']
    => org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke method 'public java.lang.String org.flywaydb.core.internal.configuration.extensions.PrepareScriptFilenameConfigurationExtension.getUndoFilename()'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

  {
    "type": "org.flywaydb.core.internal.configuration.extensions.PrepareScriptFilenameConfigurationExtension",
    "methods": [
      {
        "name": "getUndoFilename",
        "parameterTypes": []
      }
    ]
  }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
       java.base@25.0.3/java.lang.reflect.Method.acquireMethodAccessor(Method.java:149)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:562)
       com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:688)
       com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:760)
       com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
       com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:503)
       com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:342)
       com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4859)
       com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:4079)
       [...]

Test run finished after 4628 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         1 tests found           ]
[         0 tests skipped         ]
[         1 tests started         ]
[         0 tests aborted         ]
[         0 tests successful      ]
[         1 tests failed          ]


> Task :nativeTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/org.flywaydb.flyway-sqlserver_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.flywaydb:flyway-sqlserver:11.8.1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 11.12.0 at 2026-06-06T01:41:39Z with exit code 1 after 122s
❌ ERROR: 'native-image run' for 11.12.0 failed with exit code 1.
FAILED[native-image run][11.12.0][env GVM_TCK_LV=11.12.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.flywaydb:flyway-sqlserver:11.8.1]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=11.12.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.flywaydb:flyway-sqlserver:11.8.1`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27047956571
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
alvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.awaitility/awaitility/4.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest/2.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.flywaydb/flyway-sqlserver/11.8.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.flywaydb/flyway-core/12.5.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-databind/2.15.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-annotations/2.3.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-core/2.6.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.microsoft.sqlserver/mssql-jdbc/12.2.0.jre11 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:+AddAllCharsets@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=78beca1b-411b-ca03-928a-b9267f008db0 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=3
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.flywaydb.flyway-sqlserver_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (7.1s @ 0.23GiB)
 Builder configuration:
 - Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.1.0-dev+9.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (46.4s @ 1.25GiB)
   12,725 types,  16,183 fields, and  59,410 methods found reachable
      629 types,     136 fields, and   1,906 methods registered for reflection
       63 types,      71 fields, and      55 methods registered for JNI access
      168 resource accesses registered with 322.15KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (6.6s @ 1.25GiB)
[4/8] Parsing methods...      [**]                                                                      (5.0s @ 1.34GiB)
[5/8] Inlining methods...     [***]                                                                     (2.7s @ 1.42GiB)
[6/8] Compiling methods...    [*****]                                                                  (28.4s @ 1.03GiB)
[7/8] Laying out methods...   [**]                                                                      (4.9s @ 1.35GiB)
[8/8] Creating image...       [**]                                                                      (4.7s @ 1.52GiB)
  15.91MiB (35.16%) for code area:    38,078 compilation units
  25.56MiB (56.50%) for image heap:  338,165 objects and 32 resources
   3.77MiB ( 8.33%) for other data
  45.24MiB in total image size, 42.06MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.67MiB o.g.n.runtime/com.oracle.svm.core.code              4.81MiB byte[] for code metadata
   1.08MiB com.microsoft.sqlserver.jdbc                        4.40MiB java.base/char[]
 848.50KiB java.base/sun.security.ssl                          4.37MiB byte[] for string data
 773.50KiB java.base/java.util                                 2.05MiB java.base/java.lang.String
 465.15KiB java.base/com.sun.crypto.provider                   1.56MiB java.base/java.lang.Class
 412.35KiB java.base/java.lang                                 1.36MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 407.23KiB java.base/jdk.internal.classfile.impl               1.13MiB byte[] for general heap data
 342.02KiB java.base/java.lang.invoke                        644.50KiB byte[] for reflection metadata
 293.16KiB java.base/java.util.concurrent                    493.24KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 279.79KiB java.base/sun.security.provider                   416.41KiB java.base/java.util.HashMap$Node
   9.43MiB for 457 more packages                               4.37MiB for 3229 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 24 component(s) is embedded in binary (1.27KiB). 2 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       9.4s (8.6% of total time) in 1223 GCs | Peak RSS: 2.23GiB | CPU load: 3.61
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/org.flywaydb.flyway-sqlserver_tests-tests (executable, 42.06MiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'org.flywaydb.flyway-sqlserver_tests-tests' in 1m 48s.

The build process encountered 4 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile

> Task :nativeTest
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

Starting MSSQL ...
Jun 06, 2026 1:42:22 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 0e213b13-f410-4c81-b21d-9ec5d028dd71 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:0e213b13-f410-4c81-b21d-9ec5d028dd71
Jun 06, 2026 1:42:22 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: bd372950-9095-406c-8f52-7ee9392cdbce Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:bd372950-9095-406c-8f52-7ee9392cdbce
Jun 06, 2026 1:42:22 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 44ede5c8-2a64-42fa-acc3-4615f55cb8b9 Prelogin error: host localhost port 1433 Unexpected end of prelogin response after 0 bytes read
Jun 06, 2026 1:42:23 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 8312e0fe-3283-41bb-924e-66057419901c Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:8312e0fe-3283-41bb-924e-66057419901c
Jun 06, 2026 1:42:24 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 209871ba-0a9c-45da-88c9-a1b03c14d441 Prelogin error: host localhost port 1433 Unexpected end of prelogin response after 0 bytes read
Jun 06, 2026 1:42:25 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 01e98c3a-1fd0-4fda-bd22-15452ff43469 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:01e98c3a-1fd0-4fda-bd22-15452ff43469
Jun 06, 2026 1:42:26 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 3f528495-6781-4a26-99df-0fb8b874b838 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:3f528495-6781-4a26-99df-0fb8b874b838
MSSQL started
Shutting down MSSQL
flyway.sqlserver.FlywaySqlServerTests > migrate() FAILED


Failures (1):
  JUnit Jupiter:FlywaySqlServerTests:migrate()
    MethodSource [className = 'flyway.sqlserver.FlywaySqlServerTests', methodName = 'migrate', methodParameterTypes = '']
    => org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke method 'public java.lang.String org.flywaydb.core.internal.configuration.extensions.PrepareScriptFilenameConfigurationExtension.getUndoFilename()'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

    {
      "type": "org.flywaydb.core.internal.configuration.extensions.PrepareScriptFilenameConfigurationExtension",
      "methods": [
        {
          "name": "getUndoFilename",
          "parameterTypes": []
        }
      ]
    }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
       java.base@25.0.3/java.lang.reflect.Method.acquireMethodAccessor(Method.java:151)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:562)
       com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:688)
       com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:760)
       com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
       com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:503)
       com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:342)
       com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4859)
       com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:4079)
       [...]

Test run finished after 5629 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         1 tests found           ]
[         0 tests skipped         ]
[         1 tests started         ]
[         0 tests aborted         ]
[         0 tests successful      ]
[         1 tests failed          ]


> Task :nativeTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/org.flywaydb.flyway-sqlserver_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 4s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.flywaydb:flyway-sqlserver:11.8.1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 6s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 11.12.0 at 2026-06-06T01:42:27Z with exit code 1 after 126s
❌ ERROR: 'native-image run' for 11.12.0 failed with exit code 1.
FAILED[native-image run][11.12.0][env GVM_TCK_LV=11.12.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.flywaydb:flyway-sqlserver:11.8.1]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=11.12.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.flywaydb:flyway-sqlserver:11.8.1`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27047956571
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
rk/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.awaitility/awaitility/4.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest/2.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.flywaydb/flyway-sqlserver/11.8.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.flywaydb/flyway-core/12.5.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-databind/2.15.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-annotations/2.3.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.fasterxml.jackson.core/jackson-core/2.6.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/com.microsoft.sqlserver/mssql-jdbc/12.2.0.jre11 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:FutureDefaults@user+api=all \
-H:+AddAllCharsets@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=5c740173-0d8f-bdf7-cfd0-0512a9d0b45d \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=3
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.flywaydb.flyway-sqlserver_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (7.1s @ 0.23GiB)
 Builder configuration:
 - Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.1.0-dev+9.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (50.0s @ 1.36GiB)
   12,980 types,  16,664 fields, and  60,972 methods found reachable
      647 types,     136 fields, and   1,920 methods registered for reflection
       64 types,      73 fields, and      55 methods registered for JNI access
      170 resource accesses registered with 322.29KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (6.2s @ 1.25GiB)
[4/8] Parsing methods...      [***]                                                                     (5.9s @ 1.35GiB)
[5/8] Inlining methods...     [***]                                                                     (2.9s @ 1.44GiB)
[6/8] Compiling methods...    [*****]                                                                  (31.2s @ 1.20GiB)
[7/8] Laying out methods...   [**]                                                                      (4.6s @ 1.52GiB)
[8/8] Creating image...       [**]                                                                      (4.7s @ 1.72GiB)
  16.41MiB (34.74%) for code area:    39,224 compilation units
  26.75MiB (56.63%) for image heap:  357,164 objects and 34 resources
   4.07MiB ( 8.63%) for other data
  47.23MiB in total image size, 43.94MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.71MiB o.g.n.runtime/com.oracle.svm.core.code              4.97MiB byte[] for code metadata
   1.08MiB com.microsoft.sqlserver.jdbc                        4.58MiB byte[] for string data
 864.16KiB java.base/sun.security.ssl                          4.40MiB java.base/char[]
 777.34KiB java.base/java.util                                 2.13MiB java.base/java.lang.String
 464.51KiB java.base/com.sun.crypto.provider                   1.59MiB java.base/java.lang.Class
 423.94KiB java.base/java.lang                                 1.39MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 407.20KiB java.base/jdk.internal.classfile.impl               1.13MiB byte[] for general heap data
 342.02KiB java.base/java.lang.invoke                        664.75KiB byte[] for reflection metadata
 293.81KiB java.base/java.util.concurrent                    579.60KiB heap alignment
 279.79KiB java.base/sun.security.provider                   503.20KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
   9.86MiB for 462 more packages                               4.86MiB for 3287 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 24 component(s) is embedded in binary (1.27KiB). 2 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       9.9s (8.4% of total time) in 1347 GCs | Peak RSS: 2.44GiB | CPU load: 3.66
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/org.flywaydb.flyway-sqlserver_tests-tests (executable, 43.94MiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'org.flywaydb.flyway-sqlserver_tests-tests' in 1m 56s.

The build process encountered 4 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile

> Task :nativeTest
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

Starting MSSQL ...
Jun 06, 2026 1:44:00 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 0a9f396f-fa46-4d5c-bf83-7afb0dca374f Prelogin error: host localhost port 1433 Unexpected end of prelogin response after 0 bytes read
Jun 06, 2026 1:44:00 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 73334154-eca6-4a68-913e-8122df6713b1 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:73334154-eca6-4a68-913e-8122df6713b1
Jun 06, 2026 1:44:00 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 82d97d3a-e2c2-48a9-8882-a6cedd31bb53 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:82d97d3a-e2c2-48a9-8882-a6cedd31bb53
Jun 06, 2026 1:44:01 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 70e29be3-e692-41aa-89e2-a3199875a7fb Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:70e29be3-e692-41aa-89e2-a3199875a7fb
Jun 06, 2026 1:44:01 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: bec2a5d8-0c5e-4aab-abd4-77b1970a9247 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:bec2a5d8-0c5e-4aab-abd4-77b1970a9247
Jun 06, 2026 1:44:02 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: 3b717524-6ccc-45f1-8177-9dc70543d3fc Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:3b717524-6ccc-45f1-8177-9dc70543d3fc
Jun 06, 2026 1:44:03 AM com.microsoft.sqlserver.jdbc.SQLServerConnection prelogin
WARNING: ConnectionID:1 ClientConnectionId: f3c93da2-ea49-4313-bbb5-d86e8f3c3797 Prelogin error: host localhost port 1433 Error reading prelogin response: Connection reset ClientConnectionId:f3c93da2-ea49-4313-bbb5-d86e8f3c3797
MSSQL started
Shutting down MSSQL
flyway.sqlserver.FlywaySqlServerTests > migrate() FAILED


Failures (1):
  JUnit Jupiter:FlywaySqlServerTests:migrate()
    MethodSource [className = 'flyway.sqlserver.FlywaySqlServerTests', methodName = 'migrate', methodParameterTypes = '']
    => org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke method 'public java.lang.String org.flywaydb.core.internal.configuration.extensions.PrepareScriptFilenameConfigurationExtension.getUndoFilename()'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

    {
      "type": "org.flywaydb.core.internal.configuration.extensions.PrepareScriptFilenameConfigurationExtension",
      "methods": [
        {
          "name": "getUndoFilename",
          "parameterTypes": []
        }
      ]
    }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
       java.base@25.0.3/java.lang.reflect.Method.acquireMethodAccessor(Method.java:151)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:562)
       com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:688)
       com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:760)
       com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
       com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:503)
       com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:342)
       com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4859)
       com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:4079)
       [...]

Test run finished after 4626 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         1 tests found           ]
[         0 tests skipped         ]
[         1 tests started         ]
[         0 tests aborted         ]
[         0 tests successful      ]
[         1 tests failed          ]


> Task :nativeTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/build/native/nativeTestCompile/org.flywaydb.flyway-sqlserver_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 9s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.flywaydb:flyway-sqlserver:11.8.1 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 11s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 11.12.0 at 2026-06-06T01:44:05Z with exit code 1 after 132s
❌ ERROR: 'native-image run' for 11.12.0 failed with exit code 1.
FAILED[native-image run][11.12.0][env GVM_TCK_LV=11.12.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.flywaydb:flyway-sqlserver:11.8.1]
```

--- a/metadata/org.flywaydb/flyway-sqlserver/11.12.0/reachability-metadata.json
+++ b/metadata/org.flywaydb/flyway-sqlserver/11.12.0/reachability-metadata.json
@@ -1,0 +1,697 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "com.sun.crypto.provider.GaloisCounterMode$AESGCM",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "com.sun.crypto.provider.TlsKeyMaterialGenerator",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "com.sun.crypto.provider.TlsMasterSecretGenerator",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "com.sun.crypto.provider.TlsPrfGenerator$V12",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "java.lang.Comparable"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "java.net.Socket",
+      "methods" : [
+        {
+          "name" : "setOption",
+          "parameterTypes" : [
+            "java.net.SocketOption",
+            "java.lang.Object"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.util.FeatureDetector"
+      },
+      "type" : "org.apache.commons.logging.Log"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.util.FeatureDetector"
+      },
+      "type" : "org.apache.logging.log4j.Logger"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.clean.CleanModeCommandExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.clean.database.SQLServerCleanModePlugin"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.api.migration.baseline.BaselineAppliedMigration"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.api.configuration.ClassicConfiguration"
+      },
+      "type" : "org.flywaydb.core.api.migration.baseline.BaselineMigrationConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.api.migration.baseline.BaselineMigrationConfigurationExtension",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getBaselineMigrationPrefix",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "setBaselineMigrationPrefix",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.api.migration.baseline.BaselineMigrationConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.api.migration.baseline.BaselineMigrationResolver"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.api.migration.baseline.BaselineResourceTypeProvider"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.extensibility.Plugin"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.NullFlywayTelemetryManager"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.internal.command.clean.CleanModeConfigurationExtension",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getClean",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "setClean",
+          "parameterTypes" : [
+            "org.flywaydb.core.internal.command.clean.CleanModel"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.command.clean.CleanModeConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.internal.command.clean.CleanModel"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.internal.command.clean.SchemaModel"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.internal.configuration.extensions.DeployScriptFilenameConfigurationExtension",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getScriptFilename",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "setScriptFilename",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.configuration.extensions.DeployScriptFilenameConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.internal.configuration.extensions.PrepareScriptFilenameConfigurationExtension",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getScriptFilename",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getUndoFilename",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "setScriptFilename",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        },
+        {
+          "name" : "setUndoFilename",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.configuration.extensions.PrepareScriptFilenameConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.configuration.resolvers.EnvironmentProvisionerNone"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.configuration.resolvers.EnvironmentVariableResolver"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.configuration.resolvers.PlaceholderPropertyResolver"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.database.base.TestContainersDatabaseType"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.database.h2.H2DatabaseType"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.database.sqlite.SQLiteDatabaseType"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.AuthCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.CheckCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.DeployCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.DiffCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.DiffTextCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.GenerateCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.api.configuration.ClassicConfiguration"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.LicensingConfigurationExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.LicensingConfigurationExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.ModelCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.api.configuration.ClassicConfiguration"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.PATTokenConfigurationExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.PATTokenConfigurationExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.PrepareCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.proprietaryStubs.UndoCommandExtensionStub"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.api.configuration.ClassicConfiguration"
+      },
+      "type" : "org.flywaydb.core.internal.publishing.PublishingConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.core.internal.publishing.PublishingConfigurationExtension",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "isCheckDriftOnMigrate",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "isPublishResult",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "setCheckDriftOnMigrate",
+          "parameterTypes" : [
+            "boolean"
+          ]
+        },
+        {
+          "name" : "setPublishResult",
+          "parameterTypes" : [
+            "boolean"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.publishing.PublishingConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.resource.CoreResourceTypeProvider"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.core.internal.schemahistory.BaseAppliedMigration"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.database.sqlserver.KerberosModel",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getLogin",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "setLogin",
+          "parameterTypes" : [
+            "org.flywaydb.database.sqlserver.LoginModel"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.database.sqlserver.LoginModel",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getFile",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "setFile",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.extensibility.ConfigurationExtension"
+      },
+      "type" : "org.flywaydb.database.sqlserver.SQLServerConfigurationExtension",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getClean",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getKerberos",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "setClean",
+          "parameterTypes" : [
+            "org.flywaydb.core.internal.command.clean.CleanModel"
+          ]
+        },
+        {
+          "name" : "setKerberos",
+          "parameterTypes" : [
+            "org.flywaydb.database.sqlserver.KerberosModel"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.database.sqlserver.SQLServerConfigurationExtension"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.database.sqlserver.SQLServerDatabaseType"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.database.sqlserver.fabricDataWarehouse.FabricDataWarehouseDatabaseType"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "type" : "org.flywaydb.database.sqlserver.synapse.SynapseDatabaseType"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.util.FeatureDetector"
+      },
+      "type" : "org.slf4j.Logger"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.util.ClassUtils"
+      },
+      "type" : "org.slf4j.spi.SLF4JServiceProvider"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "sun.security.provider.NativePRNG",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "java.security.SecureRandomParameters"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcTemplate"
+      },
+      "type" : "sun.security.provider.SHA2$SHA256",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "sun.security.provider.SHA2$SHA256",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.schemahistory.JdbcTableSchemaHistory"
+      },
+      "type" : "sun.security.provider.SHA2$SHA256",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.database.sqlserver.SQLServerConnection"
+      },
+      "type" : "sun.security.provider.SHA2$SHA256",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.database.sqlserver.synapse.SynapseDatabaseType"
+      },
+      "type" : "sun.security.provider.SHA2$SHA256",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "sun.security.provider.X509Factory",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "sun.security.rsa.RSAPSSSignature",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.jdbc.JdbcUtils"
+      },
+      "type" : "sun.security.ssl.SSLContextImpl$TLSContext",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.plugin.PluginRegister"
+      },
+      "glob" : "META-INF/services/org.flywaydb.core.extensibility.Plugin"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.scanner.ClasspathClassScanner"
+      },
+      "glob" : "db/callback"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.scanner.Scanner"
+      },
+      "glob" : "db/migration"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.license.VersionPrinter"
+      },
+      "glob" : "org/flywaydb/core/internal/version.txt"
+    },
+    {
+      "bundle" : "com.microsoft.sqlserver.jdbc.SQLServerResource"
+    },
+    {
+      "bundle" : "sun.util.logging.resources.logging"
+    }
+  ]
+}

--- a/metadata/org.flywaydb/flyway-sqlserver/index.json
+++ b/metadata/org.flywaydb/flyway-sqlserver/index.json
@@ -1,6 +1,23 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "11.12.0",
+    "test-version" : "11.8.1",
+    "source-code-url" : "https://repo1.maven.org/maven2/org/flywaydb/flyway-sqlserver/$version$/flyway-sqlserver-$version$-sources.jar",
+    "repository-url" : "https://github.com/flyway/flyway",
+    "test-code-url" : "N/A",
+    "documentation-url" : "https://github.com/flyway/flyway/blob/flyway-$version$/documentation/Reference/Database%20Driver%20Reference/SQL%20Server%20Database.md",
+    "description" : "Flyway SQL Server support module adds Microsoft SQL Server, Azure Synapse, and Microsoft Fabric Data Warehouse database support to Flyway. It provides the database type, parser, schema handling, and clean-mode integrations used to run Flyway migrations against those platforms.",
+    "tested-versions" : [
+      "11.12.0"
+    ],
+    "allowed-packages" : [
+      "org.flywaydb",
+      "org.flywaydb.clean",
+      "org.flywaydb.database.sqlserver"
+    ]
+  },
+  {
     "metadata-version" : "11.8.1",
     "source-code-url" : "https://repo1.maven.org/maven2/org/flywaydb/flyway-sqlserver/$version$/flyway-sqlserver-$version$-sources.jar",
     "repository-url" : "https://github.com/flyway/flyway",

--- a/stats/org.flywaydb/flyway-sqlserver/11.12.0/stats.json
+++ b/stats/org.flywaydb/flyway-sqlserver/11.12.0/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "11.12.0",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 1152,
+        "missed" : 3356,
+        "ratio" : 0.255546,
+        "total" : 4508
+      },
+      "line" : {
+        "covered" : 199,
+        "missed" : 574,
+        "ratio" : 0.257439,
+        "total" : 773
+      },
+      "method" : {
+        "covered" : 73,
+        "missed" : 161,
+        "ratio" : 0.311966,
+        "total" : 234
+      }
+    }
+  } ]
+}

--- a/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/org.flywaydb/flyway-sqlserver/11.8.1/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -1,87 +1,4 @@
 {
-  "resources" : [
-    {
-      "glob" : "db/migration/V1__create_table.sql"
-    },
-    {
-      "glob" : "db/migration/V2__alter_table.sql"
-    },
-    {
-      "condition" : {
-        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
-      },
-      "glob" : "META-INF/services/java.net.spi.URLStreamHandlerProvider"
-    },
-    {
-      "condition" : {
-        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
-      },
-      "glob" : "META-INF/services/java.time.zone.ZoneRulesProvider"
-    },
-    {
-      "condition" : {
-        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
-      },
-      "glob" : "META-INF/services/java.util.spi.ResourceBundleControlProvider"
-    },
-    {
-      "condition" : {
-        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
-      },
-      "glob" : "META-INF/services/org.assertj.core.configuration.Configuration"
-    },
-    {
-      "condition" : {
-        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
-      },
-      "glob" : "META-INF/services/org.assertj.core.presentation.Representation"
-    },
-    {
-      "condition" : {
-        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
-      },
-      "glob" : "com/microsoft/sqlserver/jdbc/SQLServerResource_en.properties"
-    },
-    {
-      "condition" : {
-        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
-      },
-      "glob" : "com/microsoft/sqlserver/jdbc/SQLServerResource_en_US.properties"
-    },
-    {
-      "condition" : {
-        "typeReached" : "org.flywaydb.core.internal.resource.classpath.ClassPathResource"
-      },
-      "glob" : "db/migration/V1__create_table.sql"
-    },
-    {
-      "condition" : {
-        "typeReached" : "org.flywaydb.core.internal.resource.classpath.ClassPathResource"
-      },
-      "glob" : "db/migration/V2__alter_table.sql"
-    },
-    {
-      "condition" : {
-        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
-      },
-      "module" : "java.base",
-      "glob" : "jdk/internal/icu/impl/data/icudt76b/nfkc.nrm"
-    },
-    {
-      "condition" : {
-        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
-      },
-      "module" : "java.logging",
-      "glob" : "sun/util/logging/resources/logging_en.properties"
-    },
-    {
-      "condition" : {
-        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
-      },
-      "module" : "java.logging",
-      "glob" : "sun/util/logging/resources/logging_en_US.properties"
-    }
-  ],
   "reflection" : [
     {
       "condition" : {
@@ -578,6 +495,89 @@
           ]
         }
       ]
+    }
+  ],
+  "resources" : [
+    {
+      "glob" : "db/migration/V1__create_table.sql"
+    },
+    {
+      "glob" : "db/migration/V2__alter_table.sql"
+    },
+    {
+      "condition" : {
+        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
+      },
+      "glob" : "META-INF/services/java.net.spi.URLStreamHandlerProvider"
+    },
+    {
+      "condition" : {
+        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
+      },
+      "glob" : "META-INF/services/java.time.zone.ZoneRulesProvider"
+    },
+    {
+      "condition" : {
+        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
+      },
+      "glob" : "META-INF/services/java.util.spi.ResourceBundleControlProvider"
+    },
+    {
+      "condition" : {
+        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
+      },
+      "glob" : "META-INF/services/org.assertj.core.configuration.Configuration"
+    },
+    {
+      "condition" : {
+        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
+      },
+      "glob" : "META-INF/services/org.assertj.core.presentation.Representation"
+    },
+    {
+      "condition" : {
+        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
+      },
+      "glob" : "com/microsoft/sqlserver/jdbc/SQLServerResource_en.properties"
+    },
+    {
+      "condition" : {
+        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
+      },
+      "glob" : "com/microsoft/sqlserver/jdbc/SQLServerResource_en_US.properties"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.resource.classpath.ClassPathResource"
+      },
+      "glob" : "db/migration/V1__create_table.sql"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.flywaydb.core.internal.resource.classpath.ClassPathResource"
+      },
+      "glob" : "db/migration/V2__alter_table.sql"
+    },
+    {
+      "condition" : {
+        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
+      },
+      "module" : "java.base",
+      "glob" : "jdk/internal/icu/impl/data/icudt76b/nfkc.nrm"
+    },
+    {
+      "condition" : {
+        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
+      },
+      "module" : "java.logging",
+      "glob" : "sun/util/logging/resources/logging_en.properties"
+    },
+    {
+      "condition" : {
+        "typeReached" : "flyway.sqlserver.FlywaySqlServerTests"
+      },
+      "module" : "java.logging",
+      "glob" : "sun/util/logging/resources/logging_en_US.properties"
     }
   ]
 }

